File: benchmark/workloads/c/tableLoop.c
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    const size_t length = 1000000;
    uint64_t *values = malloc(length * sizeof(*values));
    if (values == NULL) {
        return 1;
    }
    for (size_t index = 0; index < length; ++index) {
        values[index] = index + 1;
    }
    uint64_t total = 0;
    for (size_t index = 0; index < length; ++index) {
        total += values[index];
    }
    printf("%" PRIu64 "\n", total);
    free(values);
    return 0;
}

File: benchmark/workloads/javascript/tableLoop.js
const values = new Float64Array(1_000_000);
for (let index = 0; index < values.length; index += 1) values[index] = index + 1;
let total = 0;
for (let index = 0; index < values.length; index += 1) total += values[index];
console.log(total);
